admin/class-primary-term-admin.php

```diff
@@ -252,7 +253,44 @@ protected function get_mapped_taxonomies_for_js( $taxonomies ) {
 	}
 
 	/**
-	 * Returns an array suitable for use in the javascript.
+	 * Gets a list of taxonomies based on the passed taxonomy and checks for ancestor taxonomies.
+	 *
+	 * @param string $taxonomy The taxonomy to get the list for.
+	 *
+	 * @return array The list of taxonomies.
+	 */
+	protected function get_taxonomies( $taxonomy ) {
+		$terms = get_terms( $taxonomy, array( 'hide_empty' => false ) );
+
+		foreach ( $terms as $term ) {
+			$term->parents = $this->get_ancestors( $term->term_id );
+		}
+
+		return $terms;
+	}
+
+	/**
+	 * Gets the ancestors for the passed term.
+	 *
+	 * @param int $term_id The term ID to get the ancestors for.
+	 *
+	 * @return array The ancestor slugs.
+	 */
+	protected function get_ancestors( $term_id ) {
+		$ancestors 		= array_reverse( get_ancestors( $term_id, 'category' ), true );
```

Comment:
I would pass an array with `get_ancestors( $term_id, 'category' )` as argument to this method. This will makes it more testable.